### PR TITLE
added integration test and fixed verify

### PR DESCRIPTION
**Pull request by garethahealy**
#### Summary
Attempting to add integration tests https://github.com/sigstore/helm-sigstore/issues/208

#### Release Note
None

#### Documentation
None

### CODE REVIEW

**Comment by garethahealy** [comment]
@sabre1041 ; when I try to `verify` the packaged and uploaded chart, I get the below error:

```
entry not insertable into log: missing provenance content
```

which comes from:
- https://github.com/sigstore/rekor/blob/main/pkg/types/helm/v0.0.1/entry.go#L384-L386

any ideas what I am doing wrong? or have a I hit a bug

**Comment by garethahealy** [comment]
this works against `v0.2.0` - so presume there's a bug on `main`

$ bin/helm-sigstore verify helm-sigstore-test-0.1.0.tgz --keyring .gnupg/sigstore-secring.gpg
Chart Verified Successfully From Helm entry:

Rekor Server: https://rekor.sigstore.dev
Rekor Index: 61347294
Rekor UUID: 24296fb24b8ad77ac1f281392de9110d81aae830454c03035403d9b77f9803f3be641793719f921f

**Comment by garethahealy** [comment]
after a bit of digging, i found:
- https://github.com/sigstore/rekor/blob/main/pkg/types/entries.go#L74-L77

have switched `CreateVersionedEntry` for `UnmarshalEntry` and verify works now

tested on my fork:
- https://github.com/garethahealy/helm-sigstore/actions/runs/7410567314/job/20163171539

File: Makefile [hunk 1]
```diff
@@ -69,6 +69,20 @@ lint: golangci-lint ## Runs golangci-lint linter
 test: ## Runs go tests
 	go test ./...
 
+prereqs-integration: ## Create integration tests pre-reqs
```
Comment (cpanato) [inline Makefile:72]: can we move this to a script and not have this in the makefile

File: tests/integration.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bats
+
+#
+# Copyright 2021 The Sigstore Authors.
```
Comment (cpanato) [inline tests/integration.sh:4]:
```suggestion
# Copyright 2024 The Sigstore Authors.
```

File: tests/integration.sh [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+  [ "$status" -eq 0 ]
+}
+
+@test "verify packaged chart" {
+  cmd="bin/helm-sigstore verify helm-sigstore-test-0.1.0.tgz --keyring .gnupg/sigstore-secring.gpg"
+  run ${cmd}
+
+  echo "${cmd} : ${status} : ${output}"
+  [ "$status" -eq 0 ]
+}
```
Comment (cpanato) [inline tests/integration.sh:56]: nit: new line

**Review by cpanato — CHANGES_REQUESTED** [review]
thanks, some comments

**Comment by garethahealy** [comment]
updated with suggested changes and tested @ https://github.com/garethahealy/helm-sigstore/actions/runs/7420076759/job/20190765226

File: scripts/generate-test-chart.sh [hunk 4]
```diff
@@ -0,0 +1,24 @@
+#!/usr/bin/env sh
```
Comment (cpanato) [inline scripts/generate-test-chart.sh:1]: please use `#!/usr/bin/env bash`

File: Makefile [hunk 5]
```diff
@@ -69,6 +69,9 @@ lint: golangci-lint ## Runs golangci-lint linter
 test: ## Runs go tests
 	go test ./...
 
+integration: ## Runs integration tests
+	bats --verbose-run tests/integration.sh
```
Comment (cpanato) [inline Makefile:73]: for local run, we might want to check if bats is installed and fail if not (or install it in a custom directory)

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -27,6 +27,9 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-go-
 
+      - name: Setup Bats and bats libs
+        uses: bats-core/bats-action@e4efaa4edd28f6c43ea5bd55c0a23bdd62360100 # 1.5.4
```
Comment (cpanato) [inline .github/workflows/ci.yml:31]:
```suggestion
        uses: bats-core/bats-action@f7defb94362f2053a3e73d13086a167448ea9133 # v1.10.0
```

Comment (cpanato) [reply .github/workflows/ci.yml:31]: there is a newer release

File: .github/workflows/ci.yml [hunk 7]
```diff
@@ -38,3 +41,18 @@ jobs:
 
       - name: Test
         run: make test
+
+      - name: Create Integration Pre-Reqs
+        run: scripts/generate-test-chart.sh
+
+      - name: Upload helm-sigstore-test-0.1.0.tgz
+        uses: actions/upload-artifact@c7d193f32edcb7bfad88892161225aeda64e9392 # v4
```
Comment (cpanato) [inline .github/workflows/ci.yml:49]:
```suggestion
        uses: actions/upload-artifact@c7d193f32edcb7bfad88892161225aeda64e9392 # v4.0.0
```

File: .github/workflows/ci.yml [hunk 8]
```diff
@@ -38,3 +41,18 @@ jobs:
 
       - name: Test
         run: make test
+
+      - name: Create Integration Pre-Reqs
+        run: scripts/generate-test-chart.sh
```
Comment (cpanato) [inline .github/workflows/ci.yml:46]:
```suggestion
        run: ./scripts/generate-test-chart.sh
```

File: .github/workflows/ci.yml [hunk 9]
```diff
@@ -38,3 +41,18 @@ jobs:
...
+        uses: actions/upload-artifact@c7d193f32edcb7bfad88892161225aeda64e9392 # v4
+        with:
+          name: helm-sigstore-test-0.1.0.tgz
+          path: |
+            helm-sigstore-test-0.1.0.tgz
+            helm-sigstore-test-0.1.0.tgz.prov
+            .gnupg/sigstore-secring.gpg
+
+      - name: Integration Test
+        run: make integration
```
Comment (cpanato) [inline .github/workflows/ci.yml:58]: nit: add new line

File: .gitignore [hunk 10]
```diff
@@ -20,4 +20,9 @@ dist
...
 # OSX
 .DS_Store
-.vscode
\ No newline at end of file
+.vscode
+.idea
+
+helm-sigstore-test/
+helm-sigstore-test-0.1.0.tgz
+.gnupg/
```
Comment (cpanato) [inline .gitignore:28]: as you are changing this can you add a new line as well?

**Review by cpanato — CHANGES_REQUESTED** [review]
some comments, thanks for working on this

File: .github/workflows/ci.yml (hunk 6 above)
Comment (garethahealy) [reply .github/workflows/ci.yml]: reverted, suggested change doesnt exist - https://github.com/bats-core/bats-action/tree/1.5.4

File: Makefile (hunk 5 above)
Comment (garethahealy) [reply Makefile:74]: added a check for `bats`, is that OK?

**Pull request merged**